crates/bevy_ecs/src/resource.rs

```diff
@@ -72,4 +86,123 @@ pub use bevy_ecs_macros::Resource;
     label = "invalid `Resource`",
     note = "consider annotating `{Self}` with `#[derive(Resource)]`"
 )]
-pub trait Resource: Send + Sync + 'static {}
+pub trait Resource: Component<Mutability = Mutable> {
```

Comment:
Another thing for future work here might be to support immutable resources by removing this bound and adding `Resource<Mutability = Mutable>` to things like `ResMut` and `World::resource_mut`.